### DISCUSSION TITLE
Code that worked in 1.2.4 is now broken in the  2.2.0 version

### DISCUSSION THREAD

**Discussion by KevinGuancheDarias** [awslabs/aws-java-nio-spi-for-s3#582] [Q&A] [answered]
Hello, I don't open an issue because I'm not pretty sure if it's something I'm doing wrong, but after migrating from 1.2.4 to 2.2.0 I'm getting an error, on a code that worked before the migration

![image](https://github.com/user-attachments/assets/8f99d13c-1bd2-442d-b958-a7e31f1ef1b6)

My code is just invoking the following

```java
Files.newByteChannel(path, StandardOpenOption.READ);
```
and path is like s3://some-bucket/some-dir/some-file

The reason to migrate was that the logback.xml present in the 1.2.4 was breaking my logging config

**Comment by markjschreiber** [comment]
Looks like the problem is occurring here https://github.com/awslabs/aws-java-nio-spi-for-s3/blob/main/src/main/java/software/amazon/nio/spi/s3/S3BasicFileAttributes.java#L235-L246

Not sure why that is the case. All it is doing is a HeadObject request? I will do some digging.

In the meantime, can you try this? This example works for me and is pretty closely related to what you are doing.

```java
public class ReadAllBytes {
    public static void main(String[] args) throws IOException {
        var filePath = Paths.get(URI.create(args[0]));

        final var bytes = Files.readAllBytes(filePath);
        // assumes this is a text file
        final var data = new String(bytes, StandardCharsets.UTF_8);
        System.out.println(data);
    }
}
```

**Reply by markjschreiber** [reply]
The `Files.readAllBytes` method internally calls `Files.newByteChannel` so should be equivalent to what you are doing.

**Reply by KevinGuancheDarias** [reply]
thanks!, will try tomorrow in the morning, and provide feedback!

**Reply by KevinGuancheDarias** [reply]
Ok @markjschreiber , while this works, is not exactly the same, I need to use the channel version, because our use case is that we read "on the fly" very large 7z files, this has been working with no problems, so we know it works, we even merge some files in the same stream, and it worked.

After further debugging,  the error seems to occur when building up a S3SeekableByteChannel, there's a metadata request to fetch the size of the file (probably to be able to know when the channel should close)

I will try to compare 1.2.4 version code with 2.x.x

The strange thing is that aparrently your lib is setting an operationName
![image](https://github.com/user-attachments/assets/9f1053c7-3dc6-4b9c-8a87-82c6fac33b55)


Thanks for your time

**Reply by KevinGuancheDarias** [reply, marked as answer]
Ok, after further debugging I found the offending code, I have no idea on in what git repo this class is `S3CrtAsyncHttpClient` in 2.20.116 version of awssdk.s3 , the reason it worked 1.2.4 is because it used the v1 s3 sdk.

After updating the client to 2.29.35 it works

This is the failing code, I'm unable to find the github repo, so no idea in which awssdk version it got fixed, but anyway, it works now!

![image](https://github.com/user-attachments/assets/65801bf3-b64e-4b40-86fc-f6e59ab5d148)
Note that requestOptions  is missing the operationName
